paper/paper.md

```diff
@@ -0,0 +1,109 @@
+An example of a problem involving small-scale flow features is shown in \autoref{fig1}.
+``OceanBioME.jl`` leverages Julia's multiple dispatch and effective inline capabilities to fuse its computations directly into existing ``Oceananigans.jl`` kernels, thus maintaining ``Oceananigans.jl``'s bespoke performance, memory- and cost-efficiency on GPUs in ``OceanBioME.jl``-augmented simulations.
+
+![Here we replicate the Eady problem where a background buoyancy gradient and corresponding thermal wind generate a sub-mesoscale eddy, roughly following the setup of Taylor (2016).
+To this physical setup, we added a medium complexity (9 tracers) biogeochemical model, some of which are shown above.
+On top of this, we added particles modelling the growth of sugar kelp which are free-floating and advected by the flow, and carbon dioxide exchange from the air.
+Thanks to Julia's speed and efficiency the above model (1 km × 1 km × 100 m with 64 × 64 × 16 grid points) took about 30 minutes of computing time to simulate 10 days of evolution on an Nvidia P100 GPU. Figure made with `Makie.jl` [@makie]. \label{fig1}](eady_example.png)
+
+``OceanBioME.jl`` is built with a highly modular design that allows user control and customization.
+There are three distinct module types implemented in ``OceanBioME.jl``:
+
+- First, tracer-based ecosystem modules are formulated as a set of coupled ordinary differential equations.
+These equations can be solved by ``OceanBioME.jl`` as box models, which is particularly useful for testing.
+The same modules can be integrated by ``Oceananigans.jl`` to provide tracer-based ecosystem models.
+
+- Second, boundary modules contain sets of equations which provide information at the top and bottom of the ocean.
+For example, air-sea gas exchange modules calculate the flux of carbon dioxide and oxygen at the sea surface, while sediment modules calculate fluxes of carbon and oxygen at the seafloor.
+
+- The third module type is "biologically active" particles.
+These consist of individual-based models which are solved along particle paths and can be coupled with the tracer-based modules and physics from ``Oceananigans.jl``.
+The biologically active particles can be advected by the currents, and/or they can move according to prescribed dynamics.
+For example, migrating zooplankton or fish can be modelled with biologically active particles and ``OceanBioME.jl`` allows these to interact with tracer-based components such as phytoplankton or detritus.
+
+We provide a simple framework and utilities (such as light attenuation integration) to build the necessary components of biogeochemical models.
+With the provided models, currently a simple Nutrient-Phytoplankton-Zooplankton-Detritus [@npzd] model, an intermediate complexity model, LOBSTER [@lobster], and a higher complexity model, PISCES [@pisces]. 
```

Comment:
I couldn't find the `PISCES` implementation. Is it something that you are planning to implement?

Reply:
True!